### PR TITLE
Control initial expanded state of TOC; support URLs for items

### PR DESCRIPTION
**Pull request by longhotsummer**
This means the TOC can be collapsed when first rendered, and it makes it much easier to associate arbitrary URLs with a TOC item

### CODE REVIEW

File: core/src/components/toc-item/toc-item.tsx [hunk 1]
```diff
@@ -78,6 +78,7 @@ export class TocItem {
   render() {
     const isParent = !!this.item.children?.length;
     const showItem = !this.filteredItems || this.filteredItems.has(this.item);
+    const url = this.item.url || `#${this.item.id || ''}`;
```
Comment (goose-life) [inline core/src/components/toc-item/toc-item.tsx:81]: should the `#${ before `this.item.id` come out?

**Review by goose-life — APPROVED** [review]
lgtm, I just think there's a typo (I left a comment). 

My only question, out of interest, is where did the `id` come from before, or what changed that TOCItem now needs the optional attribute?

File: core/src/components/toc-item/toc-item.tsx (hunk 1 above)
Comment (longhotsummer) [reply core/src/components/toc-item/toc-item.tsx:81]: The old mechanism for determining the URL for a TOC item was to use the id attribute to create a link like `#chp_2`. We're now adding support for an optional `url` attribute, and falling back to the old ID if it's not there. The `#${...}\` is a javascript interpolation string equivalent to `f'#{...}'` in python. The `#` is literally just that, and the `$` indicates the interpolation inside the curlies.

Comment (goose-life) [reply core/src/components/toc-item/toc-item.tsx:81]: oh my bad, I missed the closing `}` 🤦

**Pull request merged**